### ISSUE TITLE
local files?

### ISSUE THREAD

**Issue by john-clark** [comicweb/kthoom#3]
Possibility to parse local file folder for files?

**Comment by jykae** [comment]
@john-clark Could you give example use case what you want?

**Comment by john-clark** [comment]
example: cbz files are kept in a folder on the web server. A config file could have that path which populates folder tree to the menu.

**Comment by jykae** [comment]
@codedread has been working on this recently, I just created this organization in the past for possible collaboration between people who like things related to comics.

As an author he would possibly be the person to respond about this new feature, but sounds like good idea :)

**Comment by codedread** [comment]
Hi @john-clark - I'm sorry but I still didn't quite understand your feature idea.  What do you mean by  "local files"?  kthoom already supports opening local files from your computer, so I thought at first you meant storing the files locally in the web browser (like with [this](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)).

But later you mention that files would be sitting on a web server and a configuration file that describes the layout and you point kthoom at it?

**Comment by jykae** [comment]
Agreed, those 2 comments are a bit confusing, currently Kthoom has been designed in my point of view to run only on client and comic collections can exist locally (user's computer), Google Drive, or OneDrive (not implemented yet, probably MS doesn't have enough comic lovers).

If server support is what you, @john-clark, actually wish, could you describe it to issue title & improve description for first comment?

**Comment by john-clark** [comment]
My thought is the terminology is from the server perspective, because that
is the files that are being installed and configured.

Local files -  are ones located on the server or a directory that is
mounted and symlinked, possibly put on the server by some service like
Mylar.

Remote files - are ones located on the computer connecting to the server
i.e.: where you drag and drop them on the window or connect to drop box,
these are not on the server, but uploaded to be processed by the server.

**Comment by codedread** [comment]
Updated the feature request description.  In your description, it would not make sense to specify a local server directory (Localcomicdir), since kthoom runs purely in the browser and has no knowledge of the server.

However, we could use a URL parameter to point kthoom at a web server that has some files.  For example:

https://comicweb.github.io/?serverLibrary=http://www.example.com/foo/bar/

Kthoom could then hit that URL, try to parse out what files are listed there and then let the user pick a file to load (requesting it via XMLHttpRequest).

@john-clark if you have an example URL with files that I can play with, please let me know privately and I might be able to work on this feature.

**Comment by codedread** [comment]
Btw, I've updated kthoom considerably this year to support progressively unarchiving streaming Another idea that I've been pondering recently is supporting a reading list file format in kthoom.  See  https://github.com/codedread/kthoom/issues/9

**Comment by adamhathcock** [comment]
I’d also like the ability to load a file from an arbitrary URL (baring CORS issues)

**Comment by codedread** [comment]
Hey adam, feel free to look at https://github.com/codedread/kthoom/pull/13 and review
